ddtrace/appsec/utils.py

```diff
@@ -182,3 +182,52 @@ def parse_message(msg):
         msg = email.message_from_string("MIME-Version: 1.0\nContent-Type: %s\n%s" % (content_type, body))
         return parse_message(msg)
     return {}
+
+
+def parse_response_body(raw_body):
+    import json
+
+    import xmltodict
+
+    from ddtrace.appsec._constants import SPAN_DATA_NAMES
+    from ddtrace.contrib._asm_request_content import get_waf_address
+    from ddtrace.contrib.trace_utils import _get_header_value_case_insensitive
+
+    if not raw_body:
+        return
+
+    if isinstance(raw_body, dict):
+        return raw_body
+
+    headers = get_waf_address(SPAN_DATA_NAMES.RESPONSE_HEADERS_NO_COOKIES)
+    if not headers:
+        return
+    content_type = _get_header_value_case_insensitive(
+        dict(headers),
+        "content-type",
+    )
+    if not content_type:
+        return
+
+    def access_body(bd):
+        if isinstance(bd, list) and isinstance(bd[0], (str, bytes)):
+            bd = bd[0][:0].join(bd)
+        if getattr(bd, "decode", False):
+            bd = bd.decode("UTF-8", errors="ignore")
+        if len(bd) >= 0x1000000:
```

Comment:
why 16MB? 16MB depends of the SO?

Reply:
We had to choose a limit, Nicolas proposed that value. I should give it a name.